frontend/app/src/components/helper/NftDetails.vue

```diff
@@ -51,9 +77,25 @@ const imageUrl = computed<string | null>(() => {
   return get(balanceData)?.imageUrl ?? '/assets/images/placeholder.svg';
 });
 
-const name = computed<string>(() => {
-  return get(balanceData)?.name ?? get(identifier);
+const name = computed<string | null>(() => {
+  const data = get(balanceData);
+  return data?.name || data?.collectionName || null;
```

Comment:
also when the collection is what we are falling back do you think we could extract the number (token id) and show it as `${collectionName} #${tokenId}`.

Other than that I think we are good to go.